platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/GossipFactory.java

```diff
@@ -0,0 +1,253 @@
+import com.swirlds.platform.gossip.sync.config.SyncConfig;
+import com.swirlds.platform.metrics.EventIntakeMetrics;
+import com.swirlds.platform.observers.EventObserverDispatcher;
+import com.swirlds.platform.state.EmergencyRecoveryManager;
+import com.swirlds.platform.state.SwirldStateManager;
+import com.swirlds.platform.state.signed.SignedState;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Objects;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+/**
+ * Builds the gossip engine, depending on which flavor is requested in the configuration.
+ */
+public final class GossipFactory {
+
+    private static final Logger logger = LogManager.getLogger(GossipFactory.class);
+
+    private GossipFactory() {}
+
+    /**
+     * Builds the gossip engine, depending on which flavor is requested in the configuration.
+     *
+     * @param platformContext           the platform context
+     * @param threadManager             the thread manager
+     * @param time                      the wall clock time
+     * @param crypto                    can be used to sign things
+     * @param notificationEngine        used to send notifications to the app
+     * @param addressBook               the current address book
+     * @param selfId                    this node's ID
+     * @param appVersion                the version of the app
+     * @param shadowGraph               contains non-ancient events
+     * @param emergencyRecoveryManager  handles emergency recovery
+     * @param consensusRef              a pointer to consensus
+     * @param intakeQueue               the event intake queue
+     * @param freezeManager             handles freezes
+     * @param startUpEventFrozenManager prevents event creation during startup
+     * @param swirldStateManager        manages the mutable state
+     * @param startedFromGenesis        true if this node started from a genesis state
+     * @param stateManagementComponent  manages the lifecycle of the state
+     * @param eventIntakeLambda         a method that is called when something needs to be added to the event intake
+     *                                  queue
+     * @param eventObserverDispatcher   the object used to wire event intake
+     * @param eventMapper               a data structure used to track the most recent event from each node
+     * @param eventIntakeMetrics        metrics for event intake
+     * @param eventLinker               links together events, if chatter is enabled will also buffer orphans
+     * @param updatePlatformStatus      a method that updates the platform status, when called
+     * @param loadReconnectState        a method that should be called when a state from reconnect is obtained
+     * @return the gossip engine
+     */
+    public static Gossip buildGossip(
+            @NonNull PlatformContext platformContext,
```

Comment:
Can all of these parameters be `final`?

Reply:
Yes, this was an oversight, fixed.

If I had the ability to do so, I'd change the Java language so that function parameters could only be final! 😄